### PR TITLE
Add assertXXXType helpers

### PR DESCRIPTION
**Pull request by josephsavona**
Adds helper functions for asserting that a type instance conforms to the various categories of types, e.g. `assertXXXType(type: ?GraphQLType): GraphQLXXXType`.

### CODE REVIEW

File: src/type/definition.js [hunk 1]
```diff
@@ -92,6 +110,15 @@ export type GraphQLOutputType =
     GraphQLList<GraphQLOutputType>
   >;
 
+export function assertOutputType(type: GraphQLType): GraphQLOutputType {
+  invariant(
+    isOutputType(type),
+    'GraphQL: Expected type `%s` to be a GraphQLOutputType',
+    type.name
```
Comment (leebyron) [inline src/type/definition.js]: `String(type)` - not all types are named, like list and non-null.

graphql-js's `invariant` doesn't support printf format strings - so instead this should be:

```js
`Expected type "${String(type)}" to be an output type.`
```

File: src/type/definition.js [hunk 2]
```diff
@@ -263,6 +317,10 @@ export class GraphQLScalarType {
     return parser ? parser(valueNode) : null;
   }
 
+  toJSON(): mixed {
```
Comment (leebyron) [inline src/type/definition.js]: let's do toJSON in a separate PR. The right fix for that is probably to share the method rather than call through

**Comment by josephsavona** [comment]
@leebyron addressed comments

File: src/type/definition.js [hunk 3]
```diff
@@ -64,6 +72,14 @@ export type GraphQLInputType =
     GraphQLList<GraphQLInputType>
   >;
 
+export function assertInputType(type: ?GraphQLType): GraphQLInputType {
+  invariant(
+    type != null && isInputType(type),
```
Comment (leebyron) [inline src/type/definition.js]: eqeqeq applies here.

These predicate functions accept `?Type` so you can fully lose the `type != null`

**Comment by leebyron** [comment]
🎉
